python/tvm/relay/op/contrib/tensorrt.py

```diff
@@ -808,18 +884,22 @@ def is_valid_subgraph(params, body):
                     if len(tupe_type.shape) == 0:
                         logger.info("tensorrt: scalar inputs not supported")
                         return False
-                    input_batch_sizes.append(int(tupe_type.shape[0]))
+                    if not isinstance(tupe_type.shape[0], tvm.tir.expr.Any):
+                        input_batch_sizes.append(int(tupe_type.shape[0]))
             else:
                 # Scalar inputs not allowed
                 if len(var.checked_type.shape) == 0:
                     logger.info("tensorrt: scalar inputs not supported")
                     return False
-                input_batch_sizes.append(int(var.checked_type.shape[0]))
+                if not isinstance(var.checked_type.shape[0], tvm.tir.expr.Any):
```

Comment:
By not checking the batch size here (and above for tuple type input), we open ourselves up to a possibility where we have partitioned a subgraph for TRT which cannot be executed if the inputs have different batch sizes.

Maybe we should only allow subgraphs with dynamic batch sizes if there is only one input.

This limitation only applies to implicit batch mode in TRT.

Reply:
As discussed offline, there are usually multiple dynamic inputs to the TRT graph from current OD models. Therefore, I am adding a TODO to resolve this case of different dynamic batch sizes in a future PR.